registry/client/repository.go

```diff
@@ -292,9 +292,22 @@ func (t *tags) Get(ctx context.Context, tag string) (distribution.Descriptor, er
 	if err != nil {
 		return distribution.Descriptor{}, err
 	}
+
+	req, err := http.NewRequest("HEAD", u, nil)
+	if err != nil {
+		return distribution.Descriptor{}, err
+	}
+
+	for _, t := range distribution.ManifestMediaTypes() {
+		req.Header.Add("Accept", t)
+	}
```

Comment:
This method does get a manifest descriptor from the remote to get a digest for a tag which is used in a subsequent GetByDigest request.   Without the Accept header the digest returned would be that of a converted v1 manifest (given a v2 manifest being present in the remote).